src/train/extractor/extractor.py

```diff
@@ -69,27 +70,43 @@ def extract(self, query_results, energy_components, feature_group, energy_source
         # 1. compute energy different per timestamp and concat all energy component and unit
         power_data = self.get_power_data(query_results, energy_components, energy_source)
         if power_data is None:
-            return None, None, None
+            return None, None, None, None
         power_data = drop_zero_column(power_data, power_data.columns)
         power_columns = power_data.columns
         features = FeatureGroups[FeatureGroup[feature_group]]
         # 2. separate workload and system 
         workload_features = [feature for feature in features if feature not in SYSTEM_FEATURES]
         system_features = [feature for feature in features if feature in SYSTEM_FEATURES]
+        # 2.1 separate accelerator workload form workloads
+        accelerator_features = [feature for feature in features if feature in FeatureGroups[FeatureGroup.AcceleratorOnly]]
```

Comment:
No need for these two lines (81-82)

Reply:
Actually，when fg==FeatureGroup.WorkloadOnly, I still need to separate the accelerator workload from the workloads.

Reply:
Sorry. I miss that you also add mix of FeatureGroup.AcceleratorOnly in WorkloadOnly feature. 
I think we should modify function `get_workload_feature_data` to support all kind of features instead of separate the case here for the future extension. Could you please merge the logic to read `type` label in `get_workload_feature_data` function?

Reply:
Do you mean to integrate the current logic of get_workload_feature_data into get_workload_feature_data? This requires additional returns.

Reply:
Fixed All. Thanks! And according to the third point you mentioned, perhaps I don't need to deal with type labels anymore, and there won't be any additional return values.

Reply:
Thanks for your reminder!  After some consideration, adding the logic to read type label and update features seems more appropriate.

Reply:
@Yanbo0101 
I am now pushing a PR https://github.com/sustainable-computing-io/kepler-model-server/pull/139 to export the trained models to the kepler-model-db which might have multiple conflict with the current code in the main brach. 
Please let me know if you want to have this PR merge first or need time to work on the feature extension.
This PR looks good to me if you are okay with single feature. I can merge and rebase on yours. 
Otherwise, you may need to work on rebase again.

Reply:
I would like to have this PR merge first after your review. Thanks!